### PR TITLE
Separate Profile=On and Off for perf pipeline

### PR DESCRIPTION
**Pull request by adiaaida**
In pipeline jobs, we can separate these two runs into separate runs.
Doing so decreases the time spent running windows jobs since we are not
running them in sequence. Only run Profile=On scenarios for PRs. This
should reduce the time spent running the PR job.

### CODE REVIEW

File: buildpipeline/perf-pipeline.groovy [hunk 1]
```diff
@@ -56,24 +56,32 @@ def windowsPerf(String arch, String config, String uploadString, String runType,
         bat "tests\\runtest.cmd ${config} ${arch} GenerateLayoutOnly"
 
         // We run run-xunit-perf differently for each of the different job types
+
+        String profileArg = "stopwatch"
+
+        if (isProfileOn) {
+            profileArg = "BranchMispredictions+CacheMisses+InstructionRetired"
+        }
```
Comment (jorive) [inline buildpipeline/perf-pipeline.groovy]: nit: `String profileArg = isProfileOn ? "BranchMispredictions+CacheMisses+InstructionRetired"
 : "stopwatch"`

File: buildpipeline/perf-pipeline.groovy [hunk 2]
```diff
@@ -56,24 +56,32 @@ def windowsPerf(String arch, String config, String uploadString, String runType,
...
         String runXUnitCommonArgs = "-arch ${arch} -configuration ${config} -generateBenchviewData \"%WORKSPACE%\\Microsoft.Benchview.JSONFormat\\tools\" ${uploadString} ${pgoTestFlag} -runtype ${runType} ${testEnv} -optLevel ${opt_level} -jitName ${jit} -outputdir \"%WORKSPACE%\\bin\\sandbox_logs\""
         if (scenario == 'perf') {
             String runXUnitPerfCommonArgs = "${runXUnitCommonArgs} -stabilityPrefix \"START \"CORECLR_PERF_RUN\" /B /WAIT /HIGH /AFFINITY 0x2\""
             String runXUnitPerflabArgs = "${runXUnitPerfCommonArgs} -testBinLoc bin\\tests\\${os}.${arch}.${config}\\performance\\perflab\\Perflab -library"
-            bat "tests\\scripts\\run-xunit-perf.cmd ${runXUnitPerflabArgs} -collectionFlags stopwatch"
-            bat "tests\\scripts\\run-xunit-perf.cmd ${runXUnitPerflabArgs} -collectionFlags default+BranchMispredictions+CacheMisses+InstructionRetired+gcapi"
+
+            if (isProfileOn) {
+                profileArg = "default+${profileArg}+gcapi"
+            }
```
Comment (jorive) [inline buildpipeline/perf-pipeline.groovy]: nit: `profileArg = isProfileOn ? "default+${profileArg}+gcapi"
 : ${profileArg}`

**Comment by jorive** [comment]
Small nit comments, otherwise LGTM

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test Perf Build and Test please

**Pull request merged**